libcasr/src/java.rs

```diff
@@ -0,0 +1,272 @@
+//! Java module implements `ParseStacktrace` and `Exception` traits for Java reports.
+use crate::error::*;
+use crate::exception::Exception;
+use crate::execution_class::ExecutionClass;
+use crate::stacktrace::ParseStacktrace;
+use crate::stacktrace::*;
+
+use regex::Regex;
+use std::sync::Mutex;
+
+// Variable to extract crash line from java stack trace.
+static JAVA_CRASH_LINE: Mutex<String> = Mutex::new(String::new());
+
+/// Get crash line from stack trace.
+///
+/// # Argument
+///
+/// * `trace` - stack trace as 'Stacktrace' struct
+///
+/// # Return value
+///
+/// Crash line as 'CrashLine' struct
+pub fn java_crash_line(trace: &Stacktrace) -> Result<CrashLine> {
+    let mut trace = trace.clone();
+    trace.filter();
+
+    let crash_line_st =
+        JavaStacktrace::parse_stacktrace(&[(*JAVA_CRASH_LINE.lock().unwrap()).clone()])?;
+    if trace.contains(&crash_line_st[0]) {
+        crash_line_st.crash_line()
+    } else {
+        trace.crash_line()
+    }
+}
+
+/// Structure provides an interface for processing the stack trace.
+pub struct JavaStacktrace;
+
+impl ParseStacktrace for JavaStacktrace {
+    fn extract_stacktrace(stream: &str) -> Result<Vec<String>> {
```

Comment:
We should verbally discuss this function some day.

Reply:
reworked